webcrypto/tests/generateKey/failures.js

```diff
@@ -0,0 +1,229 @@
+// This file contains an adaptation of the generateKey/failures.js implementation from
+// the W3C WebCrypto API test suite.
+//
+// Some of the function have been modified to support the k6 javascript runtime,
+// and to limit its dependency to the rest of the W3C WebCrypto API test suite internal
+// codebase.
+//
+// The original failures.js implementation is available at:
+// https://github.com/web-platform-tests/wpt/blob/0b9590a78d353217ae0bc6321ecc456f2da197ec/WebCryptoAPI/generateKey/failures.js
+
+function run_test(algorithmNames) {
+    var subtle = crypto.subtle; // Change to test prefixed implementations
+
+// These tests check that generateKey throws an error, and that
+// the error is of the right type, for a wide set of incorrect parameters.
+//
+// Error testing occurs by setting the parameter that should trigger the
+// error to an invalid value, then combining that with all valid
+// parameters that should be checked earlier by generateKey, and all
+// valid and invalid parameters that should be checked later by
+// generateKey.
+//
+// There are a lot of combinations of possible parameters for both
+// success and failure modes, resulting in a very large number of tests
+// performed.
+
+
+// Setup: define the correct behaviors that should be sought, and create
+// helper functions that generate all possible test parameters for
+// different situations.
+
+    // NOTE @oleiade: We quoted the value of the resultType property when its value is CryptoKey
```

Comment:
How is the non quoted `CryptoKey` supposed to work, if you know?

Reply:
I'm not 100% sure. However, considering the Web Platform Tests suite works under the assumption its run in a browser, they must also work under the assumption that `CryptoKey` is already exported in the top-level scope?

If I try in the Safari (webkit) web inspector to type `CryptoKey`, it seems to be natively there, without the need to import it 🤔 

```
> CryptoKey
function CryptoKey() {
    [native code]
} = $2
```

Reply:
So at least in theory if we do `globalObject.CryptoKey = require("k6/experimental/webcrypto").Cryptokey` it should work? (I am pretty sure I made a typo somewhere in the code)

If so this might be one more thing that we don't really want to change in the test script but instead fix in the code runnign the test, by setting the environment as the test expect.